### DISCUSSION TITLE
adding a --no-headers flag to the flux get commands

### DISCUSSION THREAD

**Discussion by jack-evans** [fluxcd/flux2#1562] [Proposals] [answered]
this is similar to what kubectl command line exposes. Currently in flux, you get the following:
```
$ flux get source git 
NAME            READY	MESSAGE                                                              	REVISION                                           	SUSPENDED
my-repo	        True 	Fetched revision: my-branch/caa83ff9b56c8c43273c72b4fec4a18552061c5f	my-branch/caa83ff9b56c8c43273c72b4fec4a18552061c5f	False
flux-system   	True 	Fetched revision: main/e65c5154a42ee495a05d621ba7afa015a0daf616     	main/e65c5154a42ee495a05d621ba7afa015a0daf616     	False
```
what I am proposing is that adding a `--no-headers` option flag to the commands under get, producing the following:
```
$ flux get source git --no-headers
my-repo	        True 	Fetched revision: my-branch/caa83ff9b56c8c43273c72b4fec4a18552061c5f	my-branch/caa83ff9b56c8c43273c72b4fec4a18552061c5f	False
flux-system   	True 	Fetched revision: main/e65c5154a42ee495a05d621ba7afa015a0daf616     	main/e65c5154a42ee495a05d621ba7afa015a0daf616     	False
```

This would then allow a user to use scripting to automate processes, an example being:
```bash
sources=$(flux get source git --no-headers | awk '{print $1}')

while IFS=$'\n' read -r my_source; do
  flux reconcile source git "${my_source}"
done <<< "${sources}"
```

If we are looking for something more generic we could use something like --output and allow the end user to go templating?

**Comment by stefanprodan** [comment, marked as answer]
I think `--no-headers`would be a good addition to the CLI, I have opened an issue for this request. Thanks @jack-evans

**Reply by vterdunov** [reply]
I believe `json` output more convenient for scripting purposes.

**Reply by jack-evans** [reply]
There is definitely a lot of avenues you can go down 🙂

I think it also depends on implementation, having output in json sounds like a `--output` rather than `--no-headers` parameter, however I do agree that there is definite benefits of having it!
